### PR TITLE
Reset chain from genesis

### PR DESCRIPTION
**Pull request by lucafavatella**
Delivers https://www.pivotaltracker.com/story/show/157217812

* Proof that preparatory changes up to, and excluding, "Reset consensus i.e. restart testnet chain from genesis" passes CI (including Python UAT tests): [CI workflow](https://circleci.com/workflow-run/68966d44-ef1a-4362-96de-967304c5bf57)
* Proof that this PR passes Python UAT tests: [CI workflow](https://circleci.com/workflow-run/bf6e0e36-98bc-41f9-a3c9-e3e366461aa5)

TODO:
* [x] Fix unit tests
* [x] Fix `make test`
* [x] Squash test fixes as "Reset consensus i.e. restart testnet chain from genesis"
* [x] Update release notes
* [x] Rebase on master and/or revisit DB folder depending on other PR merges bumping DB folder

### CODE REVIEW

**Comment by lucafavatella** [comment]
I fixed the unit and integration (`make test`) test failures, updated release notes, rebased on recent master, and force-pushed. The test failures were in the following categories:
* Some tests were assuming that the latest protocol version was at height strictly greater than genesis height. I fixed this in a couple of preparatory commits - so to minimize effort required in any future hard fork or consensus reset;
* Some tests were attempting to create spend txs without specifying payload - now mandatory;
* Mining test required nonce changed for genesis block change (consensus protocol version).

I also made aec_spend_tx:new check that specified payload is binary.

File: apps/aecore/src/aec_spend_tx.erl [hunk 1]
```diff
@@ -29,26 +29,15 @@
 -include("blocks.hrl").
 
 -define(SPEND_TX_VSN, 2).
```
Comment (gorillainduction) [inline apps/aecore/src/aec_spend_tx.erl]: Why keep the version as 2?

Comment (lucafavatella) [reply apps/aecore/src/aec_spend_tx.erl]: Addressed in new commit.

**Comment by lucafavatella** [comment]
I ran unit tests and Python UAT tests.

**Pull request merged**